packages/opencensus-web-types/scripts/copy-types.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This script copies (and lightly pathces) type definitions from
+ * the @opencensus/core package. This allows sharing types with @opencensus/core
+ * without directly depending on it as an NPM package, which is tricky because
+ * @opencensus/core pulls in various Node-specific dependencies.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const process = require('process');
+const util = require('util');
+
+const exec = util.promisify(require('child_process').exec);
+const mkdtemp = util.promisify(fs.mkdtemp);
+const readFile = util.promisify(fs.readFile);
+const writeFile = util.promisify(fs.writeFile);
+
+/** List of files in the @opencensus/core `src` folder to copy over. */
+const FILES_TO_COPY = [
+  'common/types.ts',
+  'exporters/types.ts',
+  'metrics/export/types.ts',
+  'stats/types.ts',
+  'tags/tag-map.ts',
+  'tags/types.ts',
+  'tags/validation.ts',
+  'trace/config/types.ts',
+  'trace/instrumentation/types.ts',
+  'trace/model/types.ts',
+  'trace/propagation/types.ts',
+  'trace/sampler/types.ts',
+  'trace/types.ts',
+];
+
+const OPENCENSUS_NODE_URL =
+    'http://github.com/census-instrumentation/opencensus-node';
+
+copyFiles();
+
+async function copyFiles() {
+  if (process.argv.length < 3) {
+    throw new Error(
+        'Must specify git tag of `@opencensus/core` copy types from.');
+  }
+  const openCensusNodeTag = process.argv[2];
+
+  console.log('Copying types from @opencensus/node package ...');
+
+  // Clone and checkout the @opencensus/core repo in a temp directory.
+  const tempDir = await mkdtemp(path.join(os.tmpdir(), 'opencensus-core-'));
+  console.log(`Cloning to temp directory: ${tempDir}`);
+  await execAndLog(`git clone ${OPENCENSUS_NODE_URL}`, {cwd: tempDir});
+  const srcDir =
+      path.join(tempDir, 'opencensus-node/packages/opencensus-core/src');
+  await execAndLog(`git checkout ${openCensusNodeTag}`, {cwd: srcDir});
+
+  console.log('Patching and copying type files ...');
+  const destDir = path.join(__dirname, '../src');
+
+  for (const srcFile of FILES_TO_COPY) {
+    const srcPath = path.join(srcDir, srcFile);
+    const destPath = path.join(destDir, srcFile);
+
+    console.log(`Processing ${srcFile} ...`);
+    const contents = await readFile(srcPath, {encoding: 'utf8'});
```

Comment:
Do we need to handle error conditions here? If a file is missing or moved to new location etc?

Reply:
Good thought. Yes, I added an check that throws an exception if the file does not exist. I also added a general reject handler for the `copyFiles` promise that logs whatever error occurred and exists the program with an error code. That makes it so that any other errors that might occur (file permissions, etc.) will be clearly printed.